src/libraries/restore/runtime/runtime.depproj

```diff
@@ -26,6 +26,15 @@
     <FileToExclude Include="apphost" />
   </ItemGroup>
 
+  <!-- In-tree runtime packs do not require the host, so exclude --> 
+  <ItemGroup Condition="'$(BinPlaceTestRuntimePack)' == 'true'">
+    <FileToExclude Include="dotnet" />
```

Comment:
I think you needed to add this because of: https://github.com/dotnet/runtime/pull/34662/files#r407798496

I think we should either set that property to false or just Condition the `SetupTestingHost` target to `'$(BinPlaceTestRuntimePack)' != 'true'`